diff --git a/node_modules/jsonix/jsonix.js b/node_modules/jsonix/jsonix.js
index 3e64d0a..4625f2c 100644
--- a/node_modules/jsonix/jsonix.js
+++ b/node_modules/jsonix/jsonix.js
@@ -31,7 +31,7 @@ Jsonix.Util.extend = function(destination, source) {
 
 		// REWORK
 		// Node.js
-		sourceIsEvt = typeof window !== 'undefined' && window !== null && typeof window.Event === "function" && source instanceof window.Event;
+		var sourceIsEvt = typeof window !== 'undefined' && window !== null && typeof window.Event === "function" && source instanceof window.Event;
 
 		if (!sourceIsEvt && source.hasOwnProperty && source.hasOwnProperty('toString')) {
 			destination.toString = source.toString;
